File: frontend/src/types/flight.ts
export interface FlightDto {
  flightNumber: string;
  airline: string;
  latitude: number | null;
  longitude: number | null;
  origin: string;
  destination: string;
  direction: number | null;
  flightStatus: string | null;
  flightDate: string | null;
  altitude: number | null;
  velocity: number | null;
  isLive: boolean;
  category: number | null;
  categoryDescription: string | null;
}

export interface FlightUpdateDto {
  subscriptionId: string;
  areaName: string;
  updatedAt: string;
  flightCount: number;
  flights: FlightDto[];
}

export interface SubscriptionDto {
  id: string;
  areaName: string;
  minLatitude: number;
  maxLatitude: number;
  minLongitude: number;
  maxLongitude: number;
  createdAt: string;
  lastUpdatedAt: string;
  isActive: boolean;
  updateIntervalSeconds: number;
}

File: backend/src/FlightWatch.Domain/Entities/Flight.cs
namespace FlightWatch.Domain.Entities;

public class Flight
{
    public string FlightNumber { get; set; } = string.Empty;
    public string Airline { get; set; } = string.Empty;
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public string Origin { get; set; } = string.Empty;
    public string Destination { get; set; } = string.Empty;
    public double? Direction { get; set; }
    public double? Altitude { get; set; }
    public double? Velocity { get; set; }
    public string? FlightStatus { get; set; }
    public bool IsLive { get; set; }
    public int? Category { get; set; }
}

